shell/platform/android/io/flutter/embedding/android/FlutterViewDelegate.java

```diff
@@ -0,0 +1,49 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.embedding.android;
+
+import android.app.Activity;
+import android.content.Context;
+import android.graphics.Rect;
+import android.view.Window;
+import android.view.WindowInsets;
+import androidx.annotation.RequiresApi;
+import androidx.annotation.VisibleForTesting;
+import io.flutter.Build;
+import io.flutter.util.ViewUtils;
+import java.util.Collections;
+import java.util.List;
+
+/** A delegate class that performs the task of retrieving the bounding rect values. */
+public class FlutterViewDelegate {
+  /**
+   * Return the WindowInsets object for the provided Context. Returns null if there is no associated
```

Comment:
Can you explain why this detail is relevant documentation to this method? I'm not sure I see the throughline.

Reply:
The dart docs is a hint for callers and for maintainers to help understand the behavior of the code documented. The existing dart doc, as written, helps a little bit by describing the null behavior but it could be more helpful if it describe when context might change. That is because the caller can know for their use case of FlutterViewDelegate if it matters which context to pass.

For example if the android documentation indicates that window insets will always be the same for an app regardless of which of the apps context is used it indicated to the caller that it might be safe in situations when launching another activity to use the application context. However if insets might change across activities maybe because of their window flags then it is important to use one FlutterViewDelate per activity.